lib/cc/cli/engines/engine_command.rb

```diff
@@ -12,32 +11,29 @@ def engine_name
           @engine_name ||= @args.first
         end
 
-        def parsed_yaml
-          @parsed_yaml ||= CC::Analyzer::Config.new(yaml_content)
+        def engine_present?
+          config.engines.key?(engine_name)
         end
 
-        def yaml_content
-          filesystem.read_path(CODECLIMATE_YAML).freeze
-        end
-
-        def update_yaml
-          filesystem.write_path(CODECLIMATE_YAML, parsed_yaml.to_yaml)
+        def engine_enabled?
+          engine_present? &&
+            config.engines[engine_name].enabled?
         end
 
-        def engine_present_in_yaml?
-          parsed_yaml.engine_present?(engine_name)
+        def config
+          @config ||= CC::Yaml.parse(filesystem.read_path(CODECLIMATE_YAML))
         end
 
-        def engine_enabled?
-          parsed_yaml.engine_enabled?(engine_name)
+        def write_config
+          filesystem.write_path(CODECLIMATE_YAML, config.as_json.to_yaml)
         end
 
-        def engine_exists?
-          engines_registry_list.keys.include?(engine_name)
+        def valid_engine?(name = engine_name)
+          registry.list.keys.include?(name)
```

Comment:
`registry.list[name].present?` should be a little more performant here. I realize you didn't make the change but since you're already changing this line.